app/presenters/base_error_hasher.rb

```diff
@@ -10,7 +10,7 @@ def unsanitized_hash
   end
 
   def compound_error?
-    error.respond_to?(:underlying_errors)
+    error.is_a?(CloudController::Errors::CompoundError)
```

Comment:
Based on further exploration of [#2537](https://github.com/cloudfoundry/cloud_controller_ng/pull/2537),  I can see that there is at least one error that contains underlying_errors (SubResourceError). Its still unclear to me if we ultimately care if its a compound error or if it contains underlying errors.

Reply:
From what I can tell this line is more related to the other work in https://github.com/cloudfoundry/cloud_controller_ng/pull/2537. Is it possible to remove it from this PR since i think it is muddying the intent of this change?

Reply:
You are right, this change is unrelated to the other parts of this PR, so I've removed it.

Reply:
Now I remember, why I've included it in this PR - otherwise [this](https://github.com/cloudfoundry/cloud_controller_ng/blob/87f1c855391d6fc61d8467ae0d7c49df177fbadf/spec/acceptance/async_bindings_spec.rb#L122-L154) async bindings test fails. As this test is calling a v3 endpoint, it now uses the (correct) `V3ErrorHasher`, but with a `SubResourceError`.

When the `V3ErrorHasher` is called with a `SubResourceError` it treats it as a "compound error" (due to the duck-typing) and thus expects it to contain `APIError`s (with fields `message`, `name` and `code`; methods `compound_error_payload` -> `hash_api_errors` -> `api_error_hash`). But the `SubResourceError` contains two `AsyncBindingDeletionsTriggered` errors which are `StandardError`s.

So from the `V3ErrorHasher`'s point of view, a compound error must contain errors of a specific type. Throughout the codebase `CompoundError`s are created for `ApiError`s, whereas `SubResourceError`s contain other errors.